extensions/api/data-management/contractdefinition/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/ContractDefinitionsApiControllerIntegrationTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.eclipse.dataspaceconnector.extension.jersey.CorsFilterConfiguration;
+import org.eclipse.dataspaceconnector.extension.jersey.JerseyRestService;
+import org.eclipse.dataspaceconnector.extension.jetty.JettyConfiguration;
+import org.eclipse.dataspaceconnector.extension.jetty.JettyService;
+import org.eclipse.dataspaceconnector.extension.jetty.PortMapping;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Random;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class ContractDefinitionsApiControllerIntegrationTest {
+
+    private static int port;
+    private OkHttpClient client;
+
+    @BeforeAll
+    static void prepareWebserver() {
+        port = 1024 + new Random().nextInt(10000);
```

Comment:
I've seen different methods to create port number, a test util method that does that shared by all the modules could be introduced.

Reply:
good idea. Ideally, that method would also make sure that only unoccupied ports are returned.

Reply:
I added a method `getFreePort(int)` to the `TestUtils` class in `:common:util`